atlp-rwanda/stackup2-barefoot-backend
# #171380752 one way trip request

#### What does this PR do?
Creates API endpoint to receive _one-way trip requests_ from users
#### Description of Task to be completed?
- Create a request model and its migrations
- create API endpoint at **POST /api/trips**
- Create `Authorization middleware` to extract and check requester info from `Authorization token` obtained after login
#### How should this be manually tested?
- After cloning the project repo, run `npm i` inside the cloned repo folder
- Make sure to rename .env.example to .env, 
- In .env file, provide all values for variables under `#Development database`  and `#General`
- Finally, run the following commands respectively:
`git fetch`
`git checkout ft-request-one-way-trip-171380752`
`npm i`
`npx sequelize-cli db:create`
`npx sequelize-cli db:migrate`
`npm run start`
- If the above steps have been followed correctly, you should see a new users and trips table created in your database
- To create a one way trip request, send the following data in your request body as JSON object to `POST http://<your_server_hostname>:<server_port>/api/trips` endpoint:
```json
    travelFrom: 'kigali',
    travelTo: 'Butare',
    travelDate: '2020-03-14T15:12:22.360Z',
    travelReason: 'business meeting',
    travelType: 'one-way',
    accommodation: true,
```
#### Any background context you want to provide?
You need to create an account and use the obtained token as an `authorization header` while creating _one way trip request_ 
#### What are the relevant pivotal tracker stories?
[#171380752](https://www.pivotaltracker.com/n/projects/2435482/stories/171380752)
#### Screenshots (if appropriate)
Example request:
![req](https://user-images.githubusercontent.com/33489881/76983737-949a7880-6946-11ea-98bc-87b13ebe3814.jpg)

Example response:
![res](https://user-images.githubusercontent.com/33489881/76983712-89dfe380-6946-11ea-846b-4891635197c7.jpg)
